src/telemetry.ts

```diff
@@ -619,6 +619,14 @@ export class IEventNamePropertyMapping {
          * Hash of the package name
          */
         hashedNamev2: string;
+        /**
+         * Whether the code is located in a Python file or a notebook cell.
+         */
+        source: 'pythonFile' | 'notebookCell';
+        /**
+         * Whether the package was detected in an existing file (upon open, upon save, upon close) or when it was being used during execution.
+         */
+        when: 'onExecution' | 'onOpenCloseOrSave';
```

Comment:
Helps determine whether user runs a cell with a specific package, such as `import vega`, then we know that we need to support such packages and widgets (or help prioritize such work)